setup.py

```diff
@@ -362,33 +368,34 @@ def check_pydep(importname, module):
 class build_ext(setuptools.command.build_ext.build_ext):
     def run(self):
         # report build options
-        if USE_NUMPY:
+        cmake_cache_vars = defaultdict(lambda: False, cmake.get_cmake_cache_variables())
+        if cmake_cache_vars['USE_NUMPY']:
```

Comment:
Note to myself: this command gets run AFTER we've run cmake, so the cmake cache is guaranteed to exist at this point (so we get accurate reporting).

Reply:
Good point! I've added a comment in #21792